### PR TITLE
Websocket API message encryption implemented

### LINKED ISSUE #946: WebSocket API Authorization
Add authorization to WebSocket API

### PR DESCRIPTION
**Pull request by echupriyanov**
* All messages are encrypted with AES256 encryption and encoded in base64 to transmit as WebSocket text message for easier debugging
* API key is read from file, which can be configured in config file (default: api_token.txt)

Closes #946

### CODE REVIEW

File: api/src/error.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+use failure::Fail;
+
+#[derive(Debug, Fail)]
+pub enum KeyError {
```
Comment (rtsisyk) [inline api/src/error.rs]: Looks liks a copy-paste from stegos_keychain.

File: api/src/crypto.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+// SOFTWARE.
+
+use crypto::aes::{self, KeySize};
+use crypto::symmetriccipher::SynchronousStreamCipher;
+use rand::{thread_rng, RngCore};
+use std::iter::repeat;
+
+// Encrypts the plaintext with given 32-byte key
+// returns encrypted payload with 16-byte IV prepended
+pub fn encrypt(key: &[u8], plaintext: &[u8]) -> Vec<u8> {
```
Comment (rtsisyk) [inline api/src/crypto.rs]: @dbmcclain could you please review this part?

Comment (dbmcclain) [reply api/src/crypto.rs]: Looks okay to me. I see you using a default nonce of all zeros, and you write that nonce out as the first 16 bytes of the encrypted payload. 

Potentially unnecessary to write the nonce, but maybe you have plans to use arbitrary nonces in the future?

Comment (echupriyanov) [reply api/src/crypto.rs]: No, no, no...
```
let mut nonce: Vec<u8> = repeat(0u8).take(16).collect();
gen.fill_bytes(&mut nonce[..]);
```
fills `nonce` with random stuff

File: api/src/error.rs (hunk 1 above)
Comment (echupriyanov) [reply api/src/error.rs]: Mostly it, with minor difference with decoding errors

File: api/src/crypto.rs (hunk 2 above)
Comment (dbmcclain) [reply api/src/crypto.rs]: yes - good

File: api/src/error.rs (hunk 1 above)
Comment (dbmcclain) [reply api/src/error.rs]: LGTM

**Pull request merged**